client/kafka/kafka.go

```diff
@@ -65,17 +63,36 @@ func init() {
 	prometheus.MustRegister(messageStatus)
 }
 
+// Message abstraction of a Kafka message.
+type Message struct {
+	topic string
+	body  interface{}
+	key   *string
+}
+
 // NewMessage creates a new message.
-func NewMessage(t string, b interface{}) *Message {
-	return &Message{topic: t, body: b}
+func NewMessage(topic string, body interface{}) (*Message, error) {
+	if topic == "" {
+		return nil, errTopicEmpty
+	}
+	if body == nil {
+		return nil, errBodyIsNil
+	}
+	return &Message{topic: topic, body: body}, nil
 }
 
 // NewMessageWithKey creates a new message with an associated key.
-func NewMessageWithKey(t string, b interface{}, k string) (*Message, error) {
-	if k == "" {
-		return nil, errors.New("key string can not be null")
+func NewMessageWithKey(topic string, body interface{}, key string) (*Message, error) {
+	if topic == "" {
+		return nil, errTopicEmpty
+	}
+	if body == nil {
+		return nil, errBodyIsNil
 	}
-	return &Message{topic: t, body: b, key: &k}, nil
+	if key == "" {
+		return nil, errors.New("key is empty")
```

Comment:
Let's define this error in the same way we define eg. `errBodyIsNil`

Reply:
Usually, I define errors if I need to use them more than once, otherwise, all errors should be vars. The error is only used one.